storage-provider/server/src/p2p/mod.rs

```diff
@@ -193,6 +217,56 @@ where
             _ => {}
         }
     }
+
+    #[instrument(level = "trace", skip(self))]
+    fn on_request_response_event(
+        &mut self,
+        event: RequestResponseEvent<PieceInfoRequest, PieceInfoResponse>,
+    ) {
+        match event {
+            RequestResponseEvent::Message { peer, message } => {
+                if let Message::Request {
+                    request,
+                    channel,
+                    request_id,
+                } = message
+                {
+                    trace!("Got request with id {request_id} from {peer}");
+
+                    let response = match self.index_db.get_piece_metadata(request.piece_cid) {
+                        Ok(info) => PieceInfoResponse::Found(PieceInfo { roots: info.roots }),
+                        Err(err) => {
+                            error!(?err, "error occurred while retrieving piece info");
+                            PieceInfoResponse::NotFound(request.piece_cid)
+                        }
+                    };
+
+                    if self
+                        .swarm
+                        .behaviour_mut()
+                        .request_response
+                        .send_response(channel, response)
+                        .is_err()
+                    {
+                        error!("Failed to send piece info to {peer:?}");
+                    }
+                }
```

Comment:
This would provide more info
```suggestion
					let response_result = self
                        .swarm
                        .behaviour_mut()
                        .request_response
                        .send_response(channel, response);
                    if let Err(err) = response_result {
                        error!("Failed to send piece info to {peer:?} with error: {err}");
                    }
                }
```